### ISSUE TITLE
A question on the BT loss with Batch Norm layers

### ISSUE THREAD

**Issue by HareshKarnan** [facebookresearch/barlowtwins#43]
Hi, Thanks a lot for the very clear implementation and the paper is so easy to read!. I had a quick question on the Barlow Twins loss.

 Since Barlow Twins relies on the statistics of a batch of data, if there are batch norm layers in the encoder network, is it possible that the parameters of the BN layers be updated/affected more than any other parameters in the network to optimize the BT loss? Did you experiment without any batchnorm layers in the encoder to see if that affects the learned representations?

**Comment by jingli9111** [comment]
Hi @HareshKarnan, thanks for your interest in our paper.

We tried removing BN in the projector (the MLP being discarded after pretraining) and it works about the same. See Table 5 in our paper.
But if you remove all the BN in the backbone encoder (the ones in the ResNet-50 backbone), it gets pretty bad performance.

I agree with you that since Barlow Twins relies on statistics of a batch, some type of normalization will be necessary.

Let me know if there's any other questions.

**Comment by HareshKarnan** [comment]
Hi @jingli9111 , Thanks for your quick reply!. I noticed the same behavior on my custom data when I removed batchnorm. I have a followup question : Intuitively, why does the projector network help learn a better representation output from the encoder? What happens if we delete the projector network and apply BT loss on "representations" and not "embeddings" or both ?

**Comment by jingli9111** [comment]
Hi @HareshKarnan,

There are many theoretical works on understanding the joint-embedding method dynamics. 
We actually have a paper trying to explain why we cannot use the final embedding: https://arxiv.org/abs/2110.09348
The reason is that the embedding space suffers from dimensional collapse. The embedding vectors collapse to subspace even though there is negative terms.

BT works similarly to SimCLR on this problem. If you directly apply BT on representations, it will also suffer from dimensional collapse. Our paper explained why this collapse happened. But we cannot explain why a projector can prevent such collapse happened in the hidden layer, AKA representation. This is still an open problem.

**Comment by HareshKarnan** [comment]
Thanks! That looks interesting, I'll check out your other paper.
